### PR TITLE
[feature] Created Game Manager Script + Tests + Real World Global Volume

### LINKED ISSUE #135: Make Game Manager class

### LINKED ISSUE #154: Implement Volume effects for real world

### PR DESCRIPTION
**Pull request by VadimBhirawa**
### Main changes
Created Game Manager Script to keep track of world state (Astral World or Real World), and keep track of Astral Meter accumulation. Also created Global Volume Prefab for Real World

### Tasks List
- [close #135 ]
- [close #154 ]

### Attachment
Real World Volume-min

### CODE REVIEW

File: Assets/Astralization/Scripts/Interactables/Item/Ankh.cs [hunk 1]
```diff
@@ -22,15 +22,15 @@ protected override void Awake()
...
-            RenderSettings.fogColor = col;
-        }
-        else RenderSettings.fogColor = Color.black;
+        //_inAstral = !_inAstral;
+        //if (_inAstral)
+        //{
+        //    //Color col = Utils.ColorHelper.ParseHex("#5F466A");
+        //    RenderSettings.fogColor = col;
+        //}
+        //else RenderSettings.fogColor = Color.black;
```
Comment (naufalhilmi12345) [inline Assets/Astralization/Scripts/Interactables/Item/Ankh.cs:33]: if the system is work, maybe delete this comment

File: Assets/Astralization/Scripts/Interactables/Item/Ankh.cs [hunk 2]
```diff
@@ -8,9 +8,9 @@
 public class Ankh : Item
 {
     #region Variables
-    private bool _inAstral = false; //TODO: Refactor this to GM
+    //private bool _inAstral = false; //TODO: Refactor this to GM
```
Comment (naufalhilmi12345) [inline Assets/Astralization/Scripts/Interactables/Item/Ankh.cs:11]: if the system is work, maybe delete this comment

**Review by naufalhilmi12345 — CHANGES_REQUESTED** [review]
i think the comment section code can be deleted

**Review by naufalhilmi12345 — APPROVED** [review]
Cool! LGTM

**Review by naufalhilmi12345 — APPROVED** [review]
Cool! LGTM

File: Assets/Astralization/Scripts/Interactables/Item/Ankh.cs [hunk 3]
```diff
@@ -8,9 +8,7 @@
 public class Ankh : Item
 {
     #region Variables
-    private bool _inAstral = false; //TODO: Refactor this to GM
-
-    public static event Action ChangeWorldEvent; //TODO: Refactor this to GM
+    public static event Action ChangeWorldGM; //TODO: Refactor this to GM
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/Interactables/Item/Ankh.cs:11]: comment is no longer need, that action is already for GM

File: Assets/Astralization/Scripts/Managers/GameManager.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    public bool GetWorld() { return _isInAstralWorld; }
+    #endregion
+
+    public void InvokeChangeWorld()
+    {
+        _isInAstralWorld = !_isInAstralWorld;
+        ChangeWorldEvent?.Invoke(_isInAstralWorld);
+    }
+
+    private void OnEnable()
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/Managers/GameManager.cs:30]: put enable disable before setter getter, and make it in a region as well

File: Assets/Astralization/Scripts/States/AstralMeterLogic.cs [hunk 5]
```diff
@@ -39,32 +36,43 @@ void Update()
...
-        }
-        else
+        if (state)
         {
             _constantRate = 0.083f;
         }
+        else _constantRate = 0.05f;
+    }
+
+    public float GetAstralMeter()
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/States/AstralMeterLogic.cs:68]: put setter getter after enable disable and make it in a region

**Review by raihansyahdean — CHANGES_REQUESTED** [review]
I dont see any script to toggle the VolumeReal yet. I think you havent implement it fully.

File: Assets/Astralization/Tests/StateTest/AstralWorldEntryTest.cs [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    #region Astral World
+    [UnityTest]
+    public IEnumerator AstralWorld_MoveToAstralWorld()
+    {
+        yield return new WaitWhile(() => sceneLoaded == false);
+        IAstralMeterLogic astralMeterLogic = GameObject.Find("AstralMeter").GetComponent<IAstralMeterLogic>();
+        Assert.IsTrue(astralMeterLogic.GetConstantRate() == 0.05f);
+
+        GameObject ankhOW = GameObject.Find("OverworldItems/Ankh");
+        inputTestFixture.Press(KeyboardMouseTestFixture.RegisteredInput.MoveLeft);
```
Comment (rafiftaris) [inline Assets/Astralization/Tests/StateTest/AstralWorldEntryTest.cs:28]: use yield return SimulateInput(), for reference see TestBase

File: Assets/Astralization/Tests/StateTest/AstralWorldEntryTest.cs [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        float moveDuration = GetMovementDurationTowards(ankhOW.transform);
+        yield return new WaitForSeconds(moveDuration);
+        inputTestFixture.Release(KeyboardMouseTestFixture.RegisteredInput.MoveLeft);
+
+        inputTestFixture.Press(KeyboardMouseTestFixture.RegisteredInput.PickItem);
```
Comment (rafiftaris) [inline Assets/Astralization/Tests/StateTest/AstralWorldEntryTest.cs:33]: use yield return SimulateInput(), for reference see TestBase

File: Assets/Astralization/Tests/StateTest/AstralWorldEntryTest.cs [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        yield return null;
+        GameObject ankh = player.transform.Find("Rotate/InteractZone/Ankh").gameObject;
+        Assert.NotNull(ankh);
+
+        IInventory inventory = player.transform.Find("Rotate/InteractZone").GetComponent<IInventory>();
+        Assert.AreEqual(1, inventory.GetNumOfItem());
+        Assert.NotNull(inventory.GetActiveItem());
+        Assert.AreEqual(0, inventory.GetActiveIdx());
+
+        inputTestFixture.Press(KeyboardMouseTestFixture.RegisteredInput.UseItem);
```
Comment (rafiftaris) [inline Assets/Astralization/Tests/StateTest/AstralWorldEntryTest.cs:43]: use yield return SimulateInput(), for reference see TestBase

**Review by rafiftaris — CHANGES_REQUESTED** [review]
Assets/Astralization/Scripts/States/AstralMeterLogic.cs

- Rename Update() with another name, so the built-in Update() won't be overriden
Screen Shot 2022-02-01 at 08 56 41

**Review by rafiftaris — APPROVED** [review]

**Review by raihansyahdean — APPROVED** [review]
Finally, good job Vadim!!!

**Pull request merged**